server/tserver/src/main/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystem.java

```diff
@@ -420,91 +420,99 @@ protected Status replicateLogs(ClientContext peerContext, final HostAndPort peer
-                throw (TableNotFoundException) e;
-              } else if (e instanceof AccumuloSecurityException) {
-                throw (AccumuloSecurityException) e;
-              } else if (e instanceof AccumuloException) {
-                throw (AccumuloException) e;
-              } else {
-                throw new RuntimeException("Received unexpected exception", e);
               }
+            } else {
+              helper.recordNewStatus(p, currentStatus, target);
             }
-          } else {
-            helper.recordNewStatus(p, currentStatus, target);
+          } catch (TableNotFoundException e) {
+            log.error("Tried to update status in replication table for {} as {}, but the table did not exist", p, ProtobufUtil.toString(currentStatus), e);
+            throw new RuntimeException("Replication table did not exist, will retry", e);
+          } finally {
+            span.stop();
           }
-        } catch (TableNotFoundException e) {
-          log.error("Tried to update status in replication table for {} as {}, but the table did not exist", p, ProtobufUtil.toString(currentStatus), e);
-          throw new RuntimeException("Replication table did not exist, will retry", e);
-        } finally {
-          span.stop();
-        }
 
-        log.debug("Recorded updated status for {}: {}", p, ProtobufUtil.toString(currentStatus));
+          log.debug("Recorded updated status for {}: {}", p, ProtobufUtil.toString(currentStatus));
 
-        // If we don't have any more work, just quit
-        if (!StatusUtil.isWorkRequired(currentStatus)) {
-          return currentStatus;
+          // If we don't have any more work, just quit
+          if (!StatusUtil.isWorkRequired(currentStatus)) {
+            return currentStatus;
+          } else {
+            // Otherwise, let it loop and replicate some more data
+            lastStatus = currentStatus;
+          }
         } else {
-          // Otherwise, let it loop and replicate some more data
-          lastStatus = currentStatus;
-        }
-      } else {
-        log.debug("Did not replicate any new data for {} to {}, (state was {})", p, target, ProtobufUtil.toString(lastStatus));
+          log.debug("Did not replicate any new data for {} to {}, (state was {})", p, target, ProtobufUtil.toString(lastStatus));
 
-        // otherwise, we didn't actually replicate (likely because there was error sending the data)
-        // we can just not record any updates, and it will be picked up again by the work assigner
-        return status;
+          // otherwise, we didn't actually replicate (likely because there was error sending the data)
+          // we can just not record any updates, and it will be picked up again by the work assigner
+          return status;
+        }
+      }
+    } finally {
+      try {
+        input.close();
+      } catch (IOException e) {
+        log.warn("Received IOException trying to close input stream: {}", e);
```

Comment:
This log message should either not include the `: {}` part, or it should be have the `e` parameter twice. As is, it will convert the exception to a string to substitute for `{}` and we'd lose the stack trace.